python/src/nanoarrow/c_lib.py

```diff
@@ -257,7 +413,60 @@ def c_array_view(obj, requested_schema=None) -> CArrayView:
     return CArrayView.from_cpu_array(c_array(obj, requested_schema))
 
 
-def allocate_c_schema():
+def c_buffer(obj, requested_schema=None) -> CBuffer:
```

Comment:
Ah, I see we used `requested_schema` in `c_array(..)` before as well. Personally, for places where the _user_ would provide a schema, I would go for `schema` instead of `requested_schema` (like `c_array_from_buffers` uses)